src/trajectory/TrajectoryGenerator.java

```diff
@@ -0,0 +1,69 @@
+package trajectory;
+
+import trajectory.Segment.*;
+
+public class TrajectoryGenerator{
+
+    double maxV;
+    double maxA;
+    double controlLoop;
+
+    public TrajectoryGenerator(double maxV, double maxA, double controlLoop){
+        this.maxV = maxV;
+        this.maxA = maxA;
+        this.controlLoop = controlLoop;
+    }
+
+    /**
+     *
+     * Using the third kinematic equation we calculate the max velocity the robt can travel based 
+     * off of the desired distance we want to travel. Then we compare this calculated velocity 
+     * to the max velocity that the robot can actually do and returns the minimum value.
+     *
+     * @param   initialV    initial velocity of the robot
+     * @param   totalDistance desired distance of travel for the robot
+     * @return      the minimum between adjustedMaxV and maxV
+     *
+     */
+    private double calcAdjustedMaxV(double initialV, double totalDistance){ 
+        double adjustedMaxV = Math.sqrt((initialV*initialV)+(2*this.maxA*totalDistance));
+
+        return Math.min(adjustedMaxV, this.maxV);
+    }
+
+
```

Comment:
Notice how you're using some of the same kinematics equations again and again?
This is a case where you want to be [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself).
It might be helpful to create a `Calculations` or `Physics` class.